baseline-refaster-rules/src/main/java/com/palantir/baseline/refaster/AssertjCollectionHasSizeExactly.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.refaster;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import java.util.Collection;
+
+public final class AssertjCollectionHasSizeExactly<T> {
+
+    @BeforeTemplate
+    void bad1(Collection<T> things, int size) {
+        assertThat(things.size() == size).isTrue();
+    }
+
+    @BeforeTemplate
+    void bad2(Collection<T> things, int size) {
+        assertThat(things.size()).isEqualTo(size);
+    }
+
+    @AfterTemplate
+    void after(Collection<T> things, int size) {
+        assertThat(things).hasSize(size);
```

Comment:
is there any way we can make this work with assertions that have descriptions without duplicating each refaster rule?

Reply:
I think what we want to do in all of these cases is to match against the return type of `assertThat(T)` instead so that chained calls are also fixed.
i.e.
```
@AfterTemplate
void after(ListAssert<T> assert, int size) {
	assert.hasSize(size);
}
```